In [2]:
# import all libraries required
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from numpy import percentile
from scipy import stats

In [3]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [4]:
#Load the csv files into pandas dataframe
Salesdata=pd.read_csv('/content/drive/My Drive/Colab Notebooks/Sales_add.csv')
#Display the dataset
Salesdata




Month  ... Sales_After_digital_add(in $)
0    Month-1  ...                        270390
1    Month-2  ...                        223334
2    Month-3  ...                        244243
3    Month-4  ...                        231808
4    Month-5  ...                        258402
5    Month-6  ...                        256948
6    Month-7  ...                        222106
7    Month-8  ...                        230637
8    Month-9  ...                        226261
9   Month-10  ...                        193735
10  Month-11  ...                        203175
11  Month-12  ...                        276279
12  Month-13  ...                        205180
13  Month-14  ...                        253131
14  Month-15  ...                        229336
15  Month-16  ...                        187305
16  Month-17  ...                        234908
17  Month-18  ...                        191517
18  Month-19  ...                        227040
19  Month-20  ...                        212579
20  Month-21  ...                        263388
21  Month-22  ...                        243020

[22 rows x 5 columns]

In [5]:
#Summary of the dataset
Salesdata.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22 entries, 0 to 21
Data columns (total 5 columns):
 #   Column                          Non-Null Count  Dtype 
---  ------                          --------------  ----- 
 0   Month                           22 non-null     object
 1   Region                          22 non-null     object
 2   Manager                         22 non-null     object
 3   Sales_before_digital_add(in $)  22 non-null     int64 
 4   Sales_After_digital_add(in $)   22 non-null     int64 
dtypes: int64(2), object(3)
memory usage: 1008.0+ bytes


**The summary of the dataset shows that there are 22 entries and 5 features**


In [7]:
# Calculating the null values present in each column of the dataset
Salesdata.isna().sum() # or StudentsPerformance.isnull().sum() 

Month                             0
Region                            0
Manager                           0
Sales_before_digital_add(in $)    0
Sales_After_digital_add(in $)     0
dtype: int64

**No missing values are present in the dataset**




In [8]:
#Statistical description of the given dataset
Salesdata.describe().T

count           mean  ...       75%       max
Sales_before_digital_add(in $)   22.0  149239.954545  ...  157627.5  178939.0
Sales_After_digital_add(in $)    22.0  231123.727273  ...  250909.0  276279.0

[2 rows x 8 columns]

The statistical summary shows that the maximuma and minimum value of sales before digital add is 178939 and 130263 respectively. Again, maximum and minimum values of sales after digital add has increased and is 276279 ans 231123 respectively

**Qn.1 A company started to invest in digital marketing as a new way of their product
promotions.For that they collected data and decided to carry out a study on it.
● The company wishes to clarify whether there is any increase in sales after
stepping into digital marketing.
● The company needs to check whether there is any dependency between the
features “Region” and “Manager”.
Help the company to carry out their study with the help of data provided**


**The company wishes to clarify whether there is any increase in sales after 
stepping into digital marketing**

**Step1_Define the null hypothesis and alternative hypothesis:**

(Null hypothesis,H0):sales are increasing after stepping into digital marketing 

(Alternate hypothesis,Ha): Sales are not increasing after stepping into digital marketing.





**Step2_Set the criteria for a decision**
We can assume it as 0.05 if it is not given in the problem


In [9]:
Sbd=Salesdata['Sales_before_digital_add(in $)']  # Assigning variable to sales before add (sbd)
Sad=Salesdata['Sales_After_digital_add(in $)']   # Assigning variable to sales after  add (sad)

In [11]:
Sbd # Disaply the values of sales before add

0     132921
1     149559
2     146278
3     152167
4     159525
5     137163
6     130625
7     131140
8     171259
9     141956
10    159339
11    178939
12    145062
13    151514
14    147463
15    177195
16    140862
17    167996
18    132135
19    152493
20    147425
21    130263
Name: Sales_before_digital_add(in $), dtype: int64

In [12]:
Sad       # Display the values of sales after add

0     270390
1     223334
2     244243
3     231808
4     258402
5     256948
6     222106
7     230637
8     226261
9     193735
10    203175
11    276279
12    205180
13    253131
14    229336
15    187305
16    234908
17    191517
18    227040
19    212579
20    263388
21    243020
Name: Sales_After_digital_add(in $), dtype: int64

In [47]:
tvalue, pvalue = stats.ttest_ind(Sbd,Sad)# Independence values of tvalue and pvalue for ttest
stats.ttest_ind(Sbd,Sad)
tvalue,pvalue

(-12.995084451110877, 2.614368006904645e-16)

In [49]:
tcriteria = stats.t.ppf(1-0.05,40)  
tcriteria # 40 refers to the degrees of freedom

1.68385101138658

**Step-4: Make Decision_ we need to compare p value with the level of significance value and if p is less than significance level, null hypothesis will be rejected**

In [50]:
if (tvalue>tcriteria and pvalue <0.05):
    print('Reject null hypothesis:There is increase in sales after stepping to digital marketing')
else:
    print('Reject alternate hypothesis:There is no increase in sales after stepping to digital marketing')

Reject alternate hypothesis:There is no increase in sales after stepping to digital marketing


 Since alternate hypothesis is rejected here, it is clear that **"There is increse in sales after stepping to digital marketing"**

**2.The company needs to check whether there is any dependency between the features “Region” and “Manager”**

Define the null hypothesis and alternative hypothesis

(Null hypothesis,H0):There is no dependency between the features "Region" and "manager"


 (Alternate hypothesis,Ha): There is  dependency between the features "Region" and "Manager"

**Since it is asked to find out a relationship between the features "Region" and "Manager", we need to perform Chi squre test**

In [52]:
Data=Salesdata.groupby('Region')['Manager'].value_counts().unstack()  #grouping variables to perform Chi square test
Data

Manager     Manager - A  Manager - B  Manager - C
Region                                           
Region - A            4            3            3
Region - B            4            1            2
Region - C            1            3            1

In [58]:
chivalue,pvalue,dof,expvalues=stats.chi2_contingency(Data)  #Perform Chi square test of independence

In [60]:
chivalue    # Test statistics_Chi Square

3.050566893424036

In [61]:
pvalue  #p value of test data

0.5493991051158094

In [62]:
dof #Degrees of freedom

4

In [64]:
expvalues #expected observations

array([[4.09090909, 3.18181818, 2.72727273],
       [2.86363636, 2.22727273, 1.90909091],
       [2.04545455, 1.59090909, 1.36363636]])

In [66]:
chicriteria=stats.chi2.ppf(1-0.05,4) 
chicriteria     #Chi square criteria

9.487729036781154

**Checking Hypothesis**

In [71]:
if(chivalue>chicriteria and pvalue<0.05):
    print('Reject null hypothesis: There is no dependency between the features "Region" and "manager"')
else:
    print('Reject alternate hypothesis: There is dependency between the features "Region" and "Manager"')

Reject alternate hypothesis: There is dependency between the features "Region" and "Manager"


**Since alternate hypothesis is rejected it is inferred that **There is no dependency between the features "Region" and "Manager"**